v3/lints/cabf_smime_br/lint_strict_aia_contains_internal_names_test.go

```diff
@@ -19,14 +19,29 @@ func TestSMIMEStrictAIAInternalName(t *testing.T) {
 			ExpectedResult: lint.Pass,
 		},
 		{
-			Name:           "warn - aia with internal names",
+			Name:           "warn - aia with internal names in AIA OCSP ",
 			InputFilename:  "smime/aiaWithInternalNamesStrict.pem",
 			ExpectedResult: lint.Warn,
 		},
+		{
+			Name:           "warn - aia with internal names in AIA CA issuers ",
+			InputFilename:  "smime/aiaWithInternalNamesCaIssuersStrict.pem",
+			ExpectedResult: lint.Warn,
+		},
+		{
+			Name:           "warn - aia with valid names, one is ldap",
```

Comment:
I believe this version of the lint will now cover Legacy profile, which currently allows LDAP scheme to appear in the AIA. That may be unintentional or just missed. Along with that, I think if we're going to do this refactor for the Strict/MP profiles, that we should also do it for the legacy profiles which are additional lints that should be reviewed.

Reply:
In the contains_internal_names lint, it is checked whether it is a BR-SMIME certificate, the scheme part is not checked at all and only the host part of the URL is checked. The certificate containing the LDAP URL is a Pass Test. I believe the current implementation covers these apsects. Has something not been taken into account? If there is something missing or is not correct I could extend the implementation.

Reply:
Commit https://github.com/zmap/zlint/pull/777/commits/a5711df13cef54288ec3f735d5718643ee008805 covers this.